modules/dex/src/lib.rs

```diff
@@ -479,7 +481,8 @@ pub mod module {
 				Error::<T>::MustBeDisabled
 			);
 			ensure!(
-				T::Currency::total_issuance(trading_pair.dex_share_currency_id()).is_zero(),
+				T::Currency::total_issuance(trading_pair.dex_share_currency_id()).is_zero()
+					&& ProvisioningPool::<T>::iter_prefix(trading_pair).count().is_zero(),
```

Comment:
```suggestion
					&& ProvisioningPool::<T>::iter_prefix(trading_pair).next().is_none(),
```

we don't want to iterate everything just to find out it is not empty